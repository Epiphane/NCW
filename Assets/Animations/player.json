{
   "version": 1,
   "bones": {
      "root": {
         "position": [ 0, 0, 0 ],
         "rotation": [ 0, 0, 0 ],
         "bones": {
            "spine": {
               "position": [ 0, 0, 0 ],
               "rotation": [ 0, 0, 0 ],
               "bones": {
                  "torso": {
                     "position": [ 0, 0, 0 ],
                     "rotation": [ 0, 0, 0 ],
                     "bones": {
                        "shoulders": {
                           "position": [ 0, 1.1, 2 ],
                           "rotation": [ 0, 0, 0 ],
                           "bones": {
                              "left_hand": {
                                 "position": [ -6, 0, 0 ],
                                 "rotation": [ 0, 0, 0 ]
                              },
                              "right_hand": {
                                 "position": [ 6, 0, 0 ],
                                 "rotation": [ 0, 0, 0 ]
                              }
                           }
                        }
                     }
                  },
                  "head": {
                     "position": [ 0, 10, 0 ],
                     "rotation": [ 0, 0, 0 ],
                     "bones": {
                        "hair": {
                           "position": [ 0, 0, 0 ],
                           "rotation": [ 0, 0, 0 ]
                        }
                     }
                  }
               }
            },
            "hips": {
               "position": [ 0, 0, 0.5 ],
               "rotation": [ 0, 0, 0 ],
               "bones": {
                  "left_leg": {
                     "position": [ -2.5, 0, 0 ],
                     "rotation": [ 0, 0, 0 ],
                     "bones": {
                        "left_foot": {
                           "position": [ 0, -5.5, 0 ],
                           "rotation": [ 0, 0, 0 ]
                        }
                     }
                  },
                  "right_leg": {
                     "position": [ 3.5, 0, 0 ],
                     "rotation": [ 0, 0, 0 ],
                     "bones": {
                        "right_foot": {
                           "position": [ 0, -5.5, 0 ],
                           "rotation": [ 0, 0, 0 ]
                        }
                     }
                  }
               }
            }
         }
      }
   },
   "default": "idle",
   "states": [
      {
         "name": "idle",
         "length": 2,
         "keyframes": [
            {
               "time": 0
            }
         ]
      },
      {
         "name": "walk",
         "length": 0.8,
         "keyframes": [
            {
               "time": 0,
               "bones": {
                  "head": {
                     "position": [ 0, 10.5, 0 ],
                     "rotation": [ -5, 0, 0 ]
                  },
                  "spine": { "rotation": [ 5, 0, 0 ] },
                  "torso": { "position": [ 0, 0.5, 0 ] }
               }
            },
            {
               "time": 0.2,
               "bones": {
                  "head": {
                     "position": [ 0, 11.5, 0 ],
                     "rotation": [ -5, 0, 0 ]
                  },
                  "spine": {
                     "rotation": [ 5, 0, 0 ]
                  },
                  "torso": {
                     "position": [ 0, 1.5, 0 ],
                     "rotation": [ 0, -10, 0 ]
                  },
                  "left_leg": { "rotation": [ -45, 0, 0 ] },
                  "right_leg": { "rotation": [ 45, 0, 0 ] },
                  "left_hand": { "position": [ -5.6, 0, -2 ] },
                  "right_hand": { "position": [ 5.6, 0, 2 ] }
               }
            },
            {
               "time": 0.4,
               "bones": {
                  "head": {
                     "position": [ 0, 10.5, 0 ],
                     "rotation": [ -5, 0, 0 ]
                  },
                  "spine": { "rotation": [ 5, 0, 0 ] },
                  "torso": { "position": [ 0, 0.5, 0 ] }
               }
            },
            {
               "time": 0.6,
               "bones": {
                  "head": {
                     "position": [ 0, 11.5, 0 ],
                     "rotation": [ -5, 0, 0 ]
                  },
                  "spine": {
                     "rotation": [ 5, 0, 0 ]
                  },
                  "torso": {
                     "position": [ 0, 1.5, 0 ],
                     "rotation": [ 0, 10, 0 ]
                  },
                  "left_leg": { "rotation": [ 45, 0, 0 ] },
                  "right_leg": { "rotation": [ -45, 0, 0 ] },
                  "left_hand": { "position": [ -5.6, 0, 2 ] },
                  "right_hand": { "position": [ 5.6, 0, -2 ] }
               }
            }
         ]
      },
      {
         "name": "run",
         "length": 0.4,
         "keyframes": [
            {
               "time": 0,
               "bones": {
                  "head": {
                     "position": [ 0, 10.5, 0 ],
                     "rotation": [ -15, 0, 0 ]
                  },
                  "spine": { "rotation": [ 15, 0, 0 ] },
                  "torso": { "position": [ 0, 0.5, 0 ] },
                  "left_hand": { "position": [ -6.4, 0, 0 ] },
                  "right_hand": { "position": [ 6.4, 0, 0 ] }
               }
            },
            {
               "time": 0.1,
               "bones": {
                  "head": {
                     "position": [ 0, 11, 0 ],
                     "rotation": [ -15, 0, 0 ]
                  },
                  "spine": {
                     "rotation": [ 15, 0, 0 ]
                  },
                  "torso": {
                     "position": [ 0, 1.5, 0 ],
                     "rotation": [ 0, -20, 0 ]
                  },
                  "left_leg": { "rotation": [ -75, 0, 0 ] },
                  "right_leg": { "rotation": [ 75, 0, 0 ] },
                  "left_hand": { "position": [ -6, 0, -3 ] },
                  "right_hand": { "position": [ 6, 0, 3 ] }
               }
            },
            {
               "time": 0.2,
               "bones": {
                  "head": {
                     "position": [ 0, 10.5, 0 ],
                     "rotation": [ -15, 0, 0 ]
                  },
                  "spine": { "rotation": [ 15, 0, 0 ] },
                  "torso": { "position": [ 0, 0.5, 0 ] },
                  "left_hand": { "position": [ -6.4, 0, 0 ] },
                  "right_hand": { "position": [ 6.4, 0, 0 ] }
               }
            },
            {
               "time": 0.3,
               "bones": {
                  "head": {
                     "position": [ 0, 11, 0 ],
                     "rotation": [ -15, 0, 0 ]
                  },
                  "spine": { "rotation": [ 15, 0, 0 ] },
                  "torso": {
                     "position": [ 0, 1.5, 0 ],
                     "rotation": [ 0, 20, 0 ]
                  },
                  "left_leg": { "rotation": [ 75, 0, 0 ] },
                  "right_leg": { "rotation": [ -75, 0, 0 ] },
                  "left_hand": { "position": [ -6, 0, 3 ] },
                  "right_hand": { "position": [ 6, 0, -3 ] }
               }
            }
         ]
      }
   ],
   "transitions": {
      "idle": [
         {
            "to": "walk",
            "time": 0.5,
            "triggers": [
               { "parameter": "speed", "gte": 1 },
               { "parameter": "speed", "lt": 8 }
            ]
         },
         {
            "to": "run",
            "time": 0.5,
            "triggers": [
               { "parameter": "speed", "gte": 8 }
            ]
         }
      ],
      "walk": [
         {
            "to": "idle",
            "time": 0.2,
            "triggers": [
               { "parameter": "speed", "lt": 1 }
            ]
         },
         {
            "to": "run",
            "time": 0.5,
            "triggers": [
               { "parameter": "speed", "gte": 8 }
            ]
         }
      ],
      "run": [
         {
            "to": "idle",
            "time": 0.2,
            "triggers": [
               { "parameter": "speed", "lt": 1 }
            ]
         },
         {
            "to": "walk",
            "time": 0.5,
            "triggers": [
               { "parameter": "speed", "gte": 1 },
               { "parameter": "speed", "lt": 8 }
            ]
         }
      ]
   }
}