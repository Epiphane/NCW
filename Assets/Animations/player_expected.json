{
   "bones": {
      "root": {
         "bones": {
            "character.Hips": {
               "bones": {
                  "left_leg": {
                     "bones": {
                        "left_foot": {
                           "position": [
                              0.0,
                              -5.5,
                              0.0
                           ],
                           "rotation": [
                              0.0,
                              0.0,
                              0.0
                           ]
                        }
                     },
                     "position": [
                        -2.5,
                        0.0,
                        0.0
                     ],
                     "rotation": [
                        0.0,
                        0.0,
                        0.0
                     ]
                  },
                  "right_leg": {
                     "bones": {
                        "right_foot": {
                           "position": [
                              0.0,
                              -5.5,
                              0.0
                           ],
                           "rotation": [
                              0.0,
                              0.0,
                              0.0
                           ]
                        }
                     },
                     "position": [
                        3.5,
                        0.0,
                        0.0
                     ],
                     "rotation": [
                        0.0,
                        0.0,
                        0.0
                     ]
                  }
               },
               "position": [
                  0.0,
                  0.0,
                  0.5
               ],
               "rotation": [
                  0.0,
                  0.0,
                  0.0
               ]
            },
            "character.Spine": {
               "bones": {
                  "character.Head": {
                     "bones": {
                        "character.Hair": {
                           "position": [
                              0.0,
                              0.0,
                              0.0
                           ],
                           "rotation": [
                              0.0,
                              0.0,
                              0.0
                           ]
                        }
                     },
                     "position": [
                        0.0,
                        10.0,
                        0.0
                     ],
                     "rotation": [
                        0.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Torso": {
                     "bones": {
                        "character.Shoulders": {
                           "bones": {
                              "left_hand": {
                                 "position": [
                                    -6.0,
                                    0.0,
                                    0.0
                                 ],
                                 "rotation": [
                                    0.0,
                                    0.0,
                                    0.0
                                 ]
                              },
                              "right_hand": {
                                 "position": [
                                    6.0,
                                    0.0,
                                    0.0
                                 ],
                                 "rotation": [
                                    0.0,
                                    0.0,
                                    0.0
                                 ]
                              }
                           },
                           "position": [
                              0.0,
                              1.1,
                              2.0
                           ],
                           "rotation": [
                              0.0,
                              0.0,
                              0.0
                           ]
                        }
                     },
                     "position": [
                        0.0,
                        0.0,
                        0.0
                     ],
                     "rotation": [
                        0.0,
                        0.0,
                        0.0
                     ]
                  }
               },
               "position": [
                  0.0,
                  0.0,
                  0.0
               ],
               "rotation": [
                  0.0,
                  0.0,
                  0.0
               ]
            }
         },
         "position": [
            0.0,
            0.0,
            0.0
         ],
         "rotation": [
            0.0,
            0.0,
            0.0
         ]
      }
   },
   "default": "idle",
   "states": [
      {
         "keyframes": [
            {
               "time": 0.0
            }
         ],
         "length": 2.0,
         "name": "idle"
      },
      {
         "keyframes": [
            {
               "bones": {
                  "character.Head": {
                     "position": [
                        0.0,
                        10.5,
                        0.0
                     ],
                     "rotation": [
                        -5.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Spine": {
                     "rotation": [
                        5.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Torso": {
                     "position": [
                        0.0,
                        0.5,
                        0.0
                     ]
                  }
               },
               "time": 0.0
            },
            {
               "bones": {
                  "character.Head": {
                     "position": [
                        0.0,
                        11.5,
                        0.0
                     ],
                     "rotation": [
                        -5.0,
                        0.0,
                        0.0
                     ]
                  },
                  "left_hand": {
                     "position": [
                        -5.6,
                        0.0,
                        -2.0
                     ]
                  },
                  "left_leg": {
                     "rotation": [
                        -45.0,
                        0.0,
                        0.0
                     ]
                  },
                  "right_hand": {
                     "position": [
                        5.6,
                        0.0,
                        2.0
                     ]
                  },
                  "right_leg": {
                     "rotation": [
                        45.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Spine": {
                     "rotation": [
                        5.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Torso": {
                     "position": [
                        0.0,
                        1.5,
                        0.0
                     ],
                     "rotation": [
                        0.0,
                        -10.0,
                        0.0
                     ]
                  }
               },
               "time": 0.2
            },
            {
               "bones": {
                  "character.Head": {
                     "position": [
                        0.0,
                        10.5,
                        0.0
                     ],
                     "rotation": [
                        -5.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Spine": {
                     "rotation": [
                     5.0,
                     0.0,
                     0.0
                     ]
                  },
                  "character.Torso": {
                     "position": [
                     0.0,
                     0.5,
                     0.0
                     ]
                  }
               },
               "time": 0.4
            },
            {
               "bones": {
                  "character.Head": {
                     "position": [
                        0.0,
                        11.5,
                        0.0
                     ],
                     "rotation": [
                        -5.0,
                        0.0,
                        0.0
                     ]
                  },
                  "left_hand": {
                     "position": [
                        -5.6,
                        0.0,
                        2.0
                     ]
                  },
                  "left_leg": {
                     "rotation": [
                        45.0,
                        0.0,
                        0.0
                     ]
                  },
                  "right_hand": {
                     "position": [
                        5.6,
                        0.0,
                        -2.0
                     ]
                  },
                  "right_leg": {
                     "rotation": [
                        -45.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Spine": {
                     "rotation": [
                        5.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Torso": {
                     "position": [
                        0.0,
                        1.5,
                        0.0
                     ],
                     "rotation": [
                        0.0,
                        10.0,
                        0.0
                     ]
                  }
               },
               "time": 0.6
            }
         ],
         "length": 0.8,
         "name": "walk"
      },
      {
         "keyframes": [
            {
               "bones": {
                  "character.Head": {
                     "position": [
                        0.0,
                        10.5,
                        0.0
                     ],
                     "rotation": [
                        -15.0,
                        0.0,
                        0.0
                     ]
                  },
                  "left_hand": {
                     "position": [
                        -6.4,
                        0.0,
                        0.0
                     ]
                  },
                  "right_hand": {
                     "position": [
                        6.4,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Spine": {
                     "rotation": [
                        15.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Torso": {
                     "position": [
                        0.0,
                        0.5,
                        0.0
                     ]
                  }
               },
               "time": 0.0
            },
            {
               "bones": {
                  "character.Head": {
                     "position": [
                        0.0,
                        11.0,
                        0.0
                     ],
                     "rotation": [
                        -15.0,
                        0.0,
                        0.0
                     ]
                  },
                  "left_hand": {
                     "position": [
                        -6.0,
                        0.0,
                        -3.0
                     ]
                  },
                  "left_leg": {
                     "rotation": [
                        -75.0,
                        0.0,
                        0.0
                     ]
                  },
                  "right_hand": {
                     "position": [
                        6.0,
                        0.0,
                        3.0
                     ]
                  },
                  "right_leg": {
                     "rotation": [
                        75.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Spine": {
                     "rotation": [
                        15.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Torso": {
                     "position": [
                        0.0,
                        1.5,
                        0.0
                     ],
                     "rotation": [
                        0.0,
                        -20.0,
                        0.0
                     ]
                  }
               },
               "time": 0.1
            },
            {
               "bones": {
                  "character.Head": {
                     "position": [
                        0.0,
                        10.5,
                        0.0
                     ],
                     "rotation": [
                        -15.0,
                        0.0,
                        0.0
                     ]
                  },
                  "left_hand": {
                     "position": [
                     -6.4,
                     0.0,
                     0.0
                     ]
                  },
                  "right_hand": {
                     "position": [
                     6.4,
                     0.0,
                     0.0
                     ]
                  },
                  "character.Spine": {
                     "rotation": [
                     15.0,
                     0.0,
                     0.0
                     ]
                  },
                  "character.Torso": {
                     "position": [
                     0.0,
                     0.5,
                     0.0
                     ]
                  }
               },
               "time": 0.2
            },
            {
               "bones": {
                  "character.Head": {
                     "position": [
                        0.0,
                        11.0,
                        0.0
                     ],
                     "rotation": [
                        -15.0,
                        0.0,
                        0.0
                     ]
                  },
                  "left_hand": {
                     "position": [
                        -6.0,
                        0.0,
                        3.0
                     ]
                  },
                  "left_leg": {
                     "rotation": [
                        75.0,
                        0.0,
                        0.0
                     ]
                  },
                  "right_hand": {
                     "position": [
                        6.0,
                        0.0,
                        -3.0
                     ]
                  },
                  "right_leg": {
                     "rotation": [
                        -75.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Spine": {
                     "rotation": [
                        15.0,
                        0.0,
                        0.0
                     ]
                  },
                  "character.Torso": {
                     "position": [
                        0.0,
                        1.5,
                        0.0
                     ],
                     "rotation": [
                        0.0,
                        20.0,
                        0.0
                     ]
                  }
               },
               "time": 0.3
            }
         ],
         "length": 0.4,
         "name": "run"
      }
   ],
   "transitions": {
      "idle": [
         {
            "time": 0.5,
            "to": "walk",
            "triggers": [
               {
                  "gte": 1.0,
                  "parameter": "speed"
               },
               {
                  "lt": 8.0,
                  "parameter": "speed"
               }
            ]
         },
         {
            "time": 0.5,
            "to": "run",
            "triggers": [
               {
                  "gte": 8.0,
                  "parameter": "speed"
               }
            ]
         }
      ],
      "run": [
         {
            "time": 0.2,
            "to": "idle",
            "triggers": [
               {
                  "lt": 1.0,
                  "parameter": "speed"
               }
            ]
         },
         {
            "time": 0.5,
            "to": "walk",
            "triggers": [
               {
                  "gte": 1.0,
                  "parameter": "speed"
               },
               {
                  "lt": 8.0,
                  "parameter": "speed"
               }
            ]
         }
      ],
      "walk": [
         {
            "time": 0.2,
            "to": "idle",
            "triggers": [
               {
                  "lt": 1.0,
                  "parameter": "speed"
               }
            ]
         },
         {
            "time": 0.5,
            "to": "run",
            "triggers": [
               {
                  "gte": 8.0,
                  "parameter": "speed"
               }
            ]
         }
      ]
   },
   "version": 1
}